### ISSUE TITLE
Invoke-WebRequest -OutLeaf

### ISSUE THREAD

**Issue by dkaszews** [PowerShell/PowerShell#20010]
### Summary of the new feature / enhancement

I would like an ability to download a file with `Invoke-WebRequest` without explicitly providing its name, same as with `curl -O`. The filename should be deduced from the uri, equivalent to `-Uri $Uri -OutFile (Split-Path $Uri)`. If the uri contains query parameters, they are either discarded (`test.txt?param=value` results in `test.txt`), or an error should be given.

Additionally, I think that `-OutLeaf` without `-PassThru` should return the full resolved path to the downloaded file, so that it can be stored or piped into `Start-Process` or `Invoke-Command` for immediate execution (useful for e.g. installers).

Similar proposals were made in:
* https://github.com/PowerShell/PowerShell/issues/17554 - Closed due high complexity
* https://github.com/PowerShell/PowerShell/issues/17553 - Closed due overzealous @msftbot 

I agree with the latter, that `-OutDirectory` could be an alternative API, so that you can get `Invoke-WebRequest -OutLeaf` with `Invoke-WebRequest -OutDirectory .`, while also allowing to download files with implicit filename to a different directory.

Opens:
* Should the filename be url-decoded? I.e. should `Invoke-WebRequest example.com/file%20with%space.txt -OutLeaf` result in `file%20with%20space.txt` or `file with space.txt`? I would say latter, `curl` says former.

**Comment by CarloToso** [comment]
Similar to https://github.com/PowerShell/PowerShell/issues/11671, partially addressed by https://github.com/PowerShell/PowerShell/pull/19007 and https://github.com/PowerShell/PowerShell/pull/19385

**Comment by dkaszews** [comment]
Does #19007 mean you can just specify a directory with `-OutFile` and it will just work as I described? I did not see anything about it in documentation.

**Comment by CarloToso** [comment]
Yes, but it's still being worked on so it could not work as expected in some cases.
